src/scripts/batch/hit_counts.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Using user query strings from the input stream,
+ * this script gets the number pathways and interactions (SIF edges).
+ * Results will be the same as or close (see comments below) to
+ * what one could get at apps.pathwaycommons.org if settings and queries are same.
+ *
+ * This script simply calls the app-ui server-side modules
+ * (does not start the nore server nor web app),
+ * which, in order, search for pathways and inferred simple
+ * interactions in the Pathway Commons backend (PC2 and Sifgraph services).
+ *
+ * @author Igor Rodchenkov
+ */
+
+const {getInteractionGraphFromPC} = require('../../server/routes/interactions/generate-interactions-json');
+const {search} = require('../../server/routes/search/search');
+const csv = require('csv');
+const {pipeline} = require('stream');
+const logger = require('../../server/logger');
+
+
+const transform = csv.transform((data, callback) => {
+  let out = [];
+  let q = data['Term'];
+  out.push(q);
+
+  // apply the same hack as done on the app-ui's client-side, in ServerAPI.search (before  going /api/search)
+  if (/^((uniprot|hgnc):\w+|ncbi:[0-9]+)$/i.test(q)) {
+    q=q.replace(/^(uniprot|ncbi|hgnc):/i,"");
+  }
+
+  // A quick-fix and note -
+  // PC's all-in-one full-text index (unlike name: and xrefid: fields) was build
+  // using Lucene StandardAnalyzer, which treats '-' (dash), etc., as wildcard when tokenizing original text,
+  // which, e.g., results in q="R-MMU-975956" matches too many and irrelevant pathways (it's like R OR MMU OR 975956).
+  // One should search like q=name:R-MMU-975956 or q=xrefid:R-MMU-975956 or q="R-MMU-975956" (i.e., quotes are part of the query),
+  // but app-ui (apps.pathwaycommons.org) app does not support lucene syntax (though the app specifically detects gene/protein names
+  // and uses them to find interactions in a separate query).
+  // I.e., a user likely wanted http://apps.pathwaycommons.org/search?q=%22R-HSA-201451%22&type=Pathway (one hit - BMP pathway)
+  // rather that http://apps.pathwaycommons.org/search?q=R-HSA-201451&type=Pathway (99 nonsense hits).
+  // So let's surround with quotes all one-word queries, like make literally use "R-MMU-975956", "Crb2", etc.
+  // (note that the CSV parser removes surrounding quotes, if any, from the input values, which is expected behavior.)
+  if (!/\s/g.test(q)) {
+    q = '"' + q + '"';
+  }
+
+  search({q})
+    .then((res) => {
+      let {genes, searchHits} = res;
+      // the number of pathways
+      out.push(searchHits.length);
+      return genes.map(geneInfo => geneInfo.geneSymbol);
+    })
+    .then((symbols) => {
+      if (symbols && symbols.length > 0) {
+        return getInteractionGraphFromPC(symbols);
+      }
+    })
+    .then((graph) => {
+      (graph && graph.edges) ? out.push(graph.edges.length) : out.push(0);
```

Comment:
In this case of one query gene, number is minimum number of edges since we cap at 25.

Reply:
This code is not about one query gene; this just counts graph.edges.length of whatever query returned (query could be any text - from the first column each row in input file - might contain a word, phrase with multiple genes, etc. that users typed at Reactome's site)